modules/webgl/src/adapter/resources/webgl-buffer.ts

```diff
@@ -135,12 +139,10 @@ export class WEBGLBuffer extends Buffer {
   }
 
   /** Read data from the buffer */
-  override async readAsync(
-    byteOffset: number = 0,
-    byteLength?: number
-  ): Promise<ArrayBuffer> {
+  override async readAsync(byteOffset: number = 0, byteLength?: number): Promise<Uint8Array> {
```

Comment:
Went ahead and switched this to `Promise<Uint8Array>`, that's already what the method actually returned -- happy to move everything to ArrayBuffer if that's preferred, though.